wdn/templates_5.0/js-src/notice.babel.js

```diff
@@ -0,0 +1,331 @@
+	slideUp: ['car'],
+};
+
+const closeButtonClasses = ['dcf-absolute', 'dcf-pin-top', 'dcf-pin-right', 'dcf-mt-3', 'dcf-mr-3' ,'dcf-btn', 'dcf-btn-tertiary', 'js-notice-toggle'];
+
+/**
+ *
+ * Functions
+ */
+// function to generate GUID or UIDs
+function uuidv4() {
+	return 'xxxxxxxx-xxxx-4xxx-yxxx-xxxxxxxxxxxx'.replace(/[xy]/g, function(c) {
+		var r = Math.random() * 16 | 0, v = c == 'x' ? r : (r & 0x3 | 0x8);
+		return v.toString(16);
+	});
+}
+
+/**
+ *
+ * @param el takes an element and move it to be first child of main
+ */
+function moveElement(el) {
+	const main = document.querySelector('main');
+	const firstChild = main.firstElementChild;
+	main.insertBefore(el, firstChild);
+}
+
+/**
+ * @purpose - permanently closes the notice element
+ * @param notice - notice to be closed
+ *
+ */
+function closeNotice(notice) {
+	notice.classList.add('dcf-notice-fixedBottom--close');
+	notice.addEventListener('transitionend', e => {
+		if (e.propertyName !== "max-height") return;
+		el.classList.add('dcf-d-none');
+		el.setAttribute('aria-hidden', 'true');
+		console.log(e);
+	});
+	localStorage.setItem(notice.id,'closed');
+}
+
+/**
+ * @purpose - collapse message when collapse button is selected
+ * @param el - notice to be closed
+ * @param closeButton - the close button associated with this notice
+ * @param title - title of notice
+ * @param message - message of notice
+ *
+ */
+function collapseExpandMessage(el, closeButton, title, message) {
+	const previousState = closeButton.getAttribute('aria-expanded') === "true" ? true : false;
+	const newState = !previousState;
+
+	closeButton.setAttribute('aria-expanded', newState);
+
+	if (previousState) {
+		// if expanded, collapse message
+		closeButton.innerText = "Expand";
+		title.classList.add('dcf-notice_title--collapse');
+		message.classList.add('dcf-notice_message--collapse');
+
+		localStorage.setItem(el.id, 'collapsed');
+	} else {
+		// if collapse, expand message
+		closeButton.innerText = "Collapse";
+		message.classList.remove('dcf-notice_message--collapse');
+		title.classList.remove('dcf-notice_title--collapse');
+
+		localStorage.setItem(el.id, 'expanded');
+	}
+}
+
+
+/**
+ * @purpose add a close button to the widget and the associated click events
+ * @param el
+ * @param isCollapsible
+ *
+ */
+function addCloseButton(el, isCollapsible) {
+	const closeButton = document.createElement('button');
+	closeButtonClasses.forEach(closeButtonclass => closeButton.classList.add(closeButtonclass));
+
+	if (isCollapsible) {
+		// if notice can be collapsed
+		const noticeTitle = el.querySelector('.js-notice-title');
+		const noticeMessage = el.querySelector('.js-notice-message');
+		const noticeMessageId = uuidv4();
+
+		closeButton.innerText = 'collapse';
+		closeButton.setAttribute('aria-expanded', 'true');
+
+		if (noticeTitle) {
+			noticeMessage.classList.add('dcf-notice__title');
+		} else {
+			console.error('Your notice is missing a title.');
+			return;
+		}
```

Comment:
If statements like these can often be simplified by failing early.

```
if (!noticeTitle) {
  console.error('Your notice is missing a title.');
  return;
}

noticeMessage.classList.add('dcf-notice__title');

```

Reply:
Done, also caught a mistake, should be noticeTitle, not noticeMessage.